### ISSUE TITLE
Player.conn vs Conn.player

### ISSUE THREAD

**Issue by rwxsu** [rwxsu/goot#5]
@Raggaer, @wgrr 
What do you think is a better approach: to have the player have a `net.Conn` or to have a wrapper struct around `net.Conn`that has a player?

**Comment by Raggaer** [comment]
I prefer to just inject what we need

**Comment by wgrr** [comment]
@rwxsu i'd prefer player to be a net.Conn, so you could pass a player in any out going connection

**Comment by rwxsu** [comment]
@wgrr do you mean that player implements the net.Conn interface? or to have a wrapper function? If we have a wrapper function, we can store a pointer to the map there aswell as the player.

**Comment by wgrr** [comment]
@rwxsu yes i meant player implementing net.Conn, if the server network objs (such as player, monster, npc, ...,) knows how to read/write/close itself, the network i/o functions could be totally agnostic, there is concerns with this, especially the interaction with map data structure, which is not clear for me yet

**Comment by gpedro** [comment]
suggestion:
```go
type TibiaConnection struct {
	Connection *net.Conn
	Player     game.Player
}
```
